src/initialization/MOM_grid_initialize.F90

```diff
@@ -1217,11 +1217,17 @@ subroutine initialize_masks(G, PF, US)
                  units="m", default=0.0, scale=m_to_Z_scale)
   call get_param(PF, mdl, "MASKING_DEPTH", mask_depth, &
                  "The depth below which to mask points as land points, for which all "//&
-                 "fluxes are zeroed out. MASKING_DEPTH is ignored if negative.", &
+                 "fluxes are zeroed out. MASKING_DEPTH needs to be smaller than MINIMUM_DEPTH", &
                  units="m", default=-9999.0, scale=m_to_Z_scale)
 
+  if (mask_depth > min_depth) then
+    mask_depth = -9999.0*m_to_Z_scale
+    call MOM_error(WARNING, "MOM_grid_init: initialize_masks "//&
+                  'MASKING_DEPTH is larger than MINIMUM_DEPTH and therefore ignored.')
+  endif
+
   Dmin = min_depth
-  if (mask_depth>=0.) Dmin = mask_depth
+  if (abs(mask_depth - (-9999.*m_to_Z_scale)) > 1.0e-10) Dmin = mask_depth
```

Comment:
This expression is not dimensionally consistent.  The right hand side of the inequality should be `... > 1.0e-10*m_to_Z_scale`, with a similar issue on line 423 of MOM_shared_initialization.F90.

Reply:
Thanks for pointing this out!
A related but probably too subtle question: What is the best practice here for comparing two float point numbers here? The purpose of this expression really should be checking if `mask_depth` is taking the default or not. I ended up using a minimum thickness (an Angstrom, which should be scaled here), but not sure if the logic actually checks out: we want to check  if the depth numbers are equal, rather than two thicknesses are considered Indistinguishable

Reply:
In this case, because we are setting a real number explicitly to a hard-coded default value, we are probably OK doing a simple comparison, although in other cases your instincts of doing a comparison between real values to within a small tolerance (but larger than roundoff) would be prudent.  The dimensional scaling would have tripped us up here because 1e-10 is not scaling with the same factor as the other values here. If we had effectively rescaled our vertical units by 2^50 (which does work), we would be using something of order a light-year as our unit for vertical heights, and 1e-10 light years is not a small number compared with the depth of the ocean!